tests/test_mps.py

```diff
@@ -0,0 +1,270 @@
+import random  # type: ignore
+import pytest
+
+import cuquantum as cq  # type: ignore
+import cupy as cp  # type: ignore
+import numpy as np  # type: ignore
+
+from pytket.circuit import Circuit  # type: ignore
+from pytket.extensions.cutensornet.mps import (
+    Tensor,
+    MPSxGate,
+    MPSxMPO,
+    simulate,
+    get_amplitude,
+    ContractionAlg,
+)
+
+
+def test_init() -> None:
+    circ = Circuit(5)
+
+    mps_gate = MPSxGate(qubits=circ.qubits)
+    with mps_gate.init_cutensornet():
+        assert mps_gate.is_valid()
+
+    mps_mpo = MPSxMPO(qubits=circ.qubits)
+    with mps_mpo.init_cutensornet():
+        assert mps_mpo.is_valid()
+
+
+def test_canonicalise() -> None:
+    np.random.seed(1)
+    circ = Circuit(5)
+
+    mps_gate = MPSxGate(qubits=circ.qubits)
+    with mps_gate.init_cutensornet():
+        # Fill up the tensors with random entries
+
+        # Leftmost tensor
+        T_d = cp.empty(shape=(4, 2), dtype=mps_gate._complex_t)
+        for i0 in range(T_d.shape[0]):
+            for i1 in range(T_d.shape[1]):
+                T_d[i0][i1] = np.random.rand() + 1j * np.random.rand()
+        mps_gate.tensors[0] = Tensor(T_d, bonds=[1, len(mps_gate)])
+
+        # Middle tensors
+        for pos in range(1, len(mps_gate) - 1):
+            T_d = cp.empty(shape=(4, 4, 2), dtype=mps_gate._complex_t)
+            for i0 in range(T_d.shape[0]):
+                for i1 in range(T_d.shape[1]):
+                    for i2 in range(T_d.shape[2]):
+                        T_d[i0][i1][i2] = np.random.rand() + 1j * np.random.rand()
+            mps_gate.tensors[pos] = Tensor(
+                T_d, bonds=[pos, pos + 1, pos + len(mps_gate)]
+            )
+
+        # Rightmost tensor
+        T_d = cp.empty(shape=(4, 2), dtype=mps_gate._complex_t)
+        for i0 in range(T_d.shape[0]):
+            for i1 in range(T_d.shape[1]):
+                T_d[i0][i1] = np.random.rand() + 1j * np.random.rand()
+        mps_gate.tensors[len(mps_gate) - 1] = Tensor(
+            T_d, bonds=[len(mps_gate) - 1, 2 * len(mps_gate) - 1]
+        )
+
+        mps_copy = mps_gate.copy()
+        # Calculate the norm of the MPS
+        norm_sq = mps_gate.vdot(mps_copy)
+
+        # Canonicalise around center_pos
+        center_pos = 2
+        mps_gate.canonicalise(l_pos=center_pos, r_pos=center_pos)
+
+        # Check that canonicalisation did not change the vector
+        overlap = mps_gate.vdot(mps_copy)
+        assert np.isclose(overlap, norm_sq)
+
+        # Check that the corresponding tensors are in orthogonal form
+        for pos in range(len(mps_gate)):

```

Comment:
I haven't thought about that... I think it's more of a pain than what's worth in the case of MPS, but it may be a good idea to simplify traversing Tree tensor networks, I'll think about it!

In terms of this PR, I'd say we skip it.